### PR TITLE
Improve archiving function

### PR DESCRIPTION
**Pull request by matthias-brun**
This pr makes it so users that don't have the appropriate permissions to archive a channel aren't shown the Archiving button but instead simply see the current state of the room. ( i.e. 'active' or 'archived')
Those users who do have the necessary permissions also get the option to edit the state, which now works the same way that editing the room type ('private', 'channel') does.
screen shot 2016-01-22 at 18 00 09
screen shot 2016-01-22 at 18 00 15

### CODE REVIEW

**Comment by marceloschmidt** [comment]
lgtm

File: packages/rocketchat-channel-settings/client/views/channelSettings.coffee [hunk 1]
```diff
@@ -106,4 +98,13 @@ Template.channelSettings.onCreated ->
 								return toastr.error TAPi18n.__(err.reason, err.details.roomType)
 							return toastr.error TAPi18n.__(err.reason)
 						toastr.success TAPi18n.__ 'Room_type_changed_successfully'
+			when 'archivationState'
+				if @$('input[name=archivationState]:checked').val() is 'true'
+					Meteor.call 'archiveRoom', @data?.rid, (err, results) ->
```
Comment (rodrigok) [inline packages/rocketchat-channel-settings/client/views/channelSettings.coffee]: Shouldn't check to not call `archiveRoom` in already archived rooms? The same for unarchiveRoom.

Comment (graywolf336) [inline packages/rocketchat-channel-settings/client/views/channelSettings.coffee]: What exactly do you mean by that @rodrigok? The first sentence contains a double negative, so I can't exactly make out what you're really saying...I assume it means something, but not 100% sure and we all know what they say about assuming things. :smile:

Comment (matthias-brun) [inline packages/rocketchat-channel-settings/client/views/channelSettings.coffee]: @graywolf336 What he means is that before calling `archiveRoom` or `unarchiveRoom` there should be a check whether it is even necessary to archive the room or if it is archived already. (i.e. If the user edits the settings of an archived channel, leaves the radio button on archived and clicks on save.)
@rodrigok I'll make sure to add that in.

Comment (rodrigok) [inline packages/rocketchat-channel-settings/client/views/channelSettings.coffee]: @graywolf336 sorry about that, I will try to explain better on the next time, fell free to correct me, I'm learning english yet :smile: 

@matthias-brun thanks :smiley:

**Comment by matthias-brun** [comment]
Okay, I added in the check we discussed.

I was also thinking it might make sense to integrate archiving in the `saveRoomSettings` method (which is used for the room name/topic/type) to ensure that behaviour is consistent (and the code is consistently structured).
But I'll leave that decision to you guys since I'm not sure about what implications that change might have.

**Comment by rodrigok** [comment]
@matthias-brun IMHO separated methods are a better solution, more clear and we can wrap for APIs like REST API.

@RocketChat/core what do you think?

**Comment by rodrigok** [comment]
LGTM

**Comment by graywolf336** [comment]
@rodrigok Yes, I agree that having different methods is better solution as I had trouble figuring out how to save a room's name after the update for the importers.

lgtm

**Pull request merged**